src/graphics/webgpu/webgpu-render-target.js

```diff
@@ -0,0 +1,202 @@
+
+    /**
+     * Unique key used by render pipeline creation
+     *
+     * @type {string}
+     */
+    key;
+
+    /** @type {string} */
+    depthFormat = 'depth24plus-stencil8';
+
+    /** @type {GPUTexture} */
+    multisampledColorBuffer;
+
+    /** @type {GPUTexture} */
+    depthTexture = null;
+
+    /**
+     * Render pass descriptor used when starting a render pass for this render target.
+     *
+     * @type {GPURenderPassDescriptor}
+     */
+    renderPassDescriptor = {};
+
+    /**
+     * @param {RenderTarget} renderTarget - The render target owning this implementation.
+     */
+    constructor(renderTarget) {
+        this.renderTarget = renderTarget;
+
+        // TODO: handle shadow map case (depth only, no color)
+
+        // key used by render pipeline creation
+        this.key = `${this.colorFormat}-${renderTarget.depth ? this.depthFormat : ''}-${renderTarget.samples}`;
+    }
+
+    /**
+     * @param {WebgpuGraphicsDevice} device - The graphics device.
+     */
+    destroy(device) {
+        this.initialized = false;
+        this.renderPassDescriptor = null;
+
+        this.depthTexture?.destroy();
+        this.depthTexture = null;
+
+        this.multisampledColorBuffer?.destroy();
+        this.multisampledColorBuffer = null;
+    }
+
+    /**
+     * Assign a color buffer. This allows the color buffer of the main framebuffer
+     * to be swapped each frame to a buffer provided by the context.
+     *
+     * @param {GPUTexture} gpuTexture - The color buffer.
+     */
+    assignColorTexture(gpuTexture) {
+
+        Debug.assert(gpuTexture);
+        const view = gpuTexture.createView();
+
+        // use it as render buffer or resolve target
+        const colorAttachment = this.renderPassDescriptor.colorAttachments[0];
+        const samples = this.renderTarget.samples;
+        if (samples > 1) {
+            colorAttachment.resolveTarget = view;
+        } else {
+            colorAttachment.view = view;
+        }
+    }
+
+    /**
+     * Initialize render target for rendering one time.
+     *
+     * @param {WebgpuGraphicsDevice} device - The graphics device.
+     * @param {RenderTarget} renderTarget - The render target.
+     */
+    init(device, renderTarget) {
+
+        Debug.assert(!this.initialized);
+        Debug.assert(!renderTarget._depthBuffer, 'WebgpuRenderTarget does not yet support options.depthBuffer');
+        const wgpu = device.wgpu;
+
+        const samples = renderTarget.samples;
+        const width = renderTarget.width;
+        const height = renderTarget.height;
+
+        // depth buffer that we render to (single or multi-sampled). We don't create resolve
+        // depth buffer as we don't currently resolve it. This might need to change in the future.
+        if (renderTarget.depth) {
+
+            /** @type {GPUTextureDescriptor} */
+            const depthTextureDesc = {
+                size: [renderTarget.width, renderTarget.height, 1],
```

Comment:
Using object desctructuring you can save a bunch of lines here and prevent multiple property lookups:

```suggestion
        const { samples, width, height, depth } = renderTarget;

        // depth buffer that we render to (single or multi-sampled). We don't create resolve
        // depth buffer as we don't currently resolve it. This might need to change in the future.
        if (depth) {

            /** @type {GPUTextureDescriptor} */
            const depthTextureDesc = {
                size: [width, height, 1],
```

Reply:
Nice tip! That is certainly better. 😄